content/en/references/configuration.md

```diff
@@ -308,40 +306,90 @@ To learn more about these configuration options, see [DNS Server]({{< ref "dns-s
 ## LocalStack Pro
 
 | Variable             | Example Values | Description |
-|----------------------|----------------| - |
+|----------------------|----------------|-------------|
 | `ACTIVATE_PRO`       | 1 (default)    | Whether pro should be activated or not. This is set to true by default if using the `localstack/localstack-pro` container image. If set to `1`, LocalStack will fail to start if the license key activation did not work. If set to `0`, an attempt is made to start LocalStack without pro features.
 | `LOCALSTACK_API_KEY` |                | API key to activate LocalStack Pro.
 | `LOG_LICENSE_ISSUES` | 1 (default)    | Whether to log issues with the license activation to the console.
 
 
 ## Deprecated
 
-These configurations are deprecated and will be removed in the upcoming major version.
+These configurations are deprecated and **will be removed in the next major version**.
 
-| Variable | Example Values | Description |
-| - | - | - |
-| `BIGDATA_MONO_CONTAINER` | `0`\|`1` (default) | **Deprecated.** Whether to spin Big Data services inside the LocalStack main container. Glue jobs breaks when using `BIGDATA_MONO_CONTAINER=0`. | 
-| `SKIP_INFRA_DOWNLOADS` | | **Deprecated.** Whether to skip downloading additional infrastructure components (e.g., specific Elasticsearch versions) |
-| `STEPFUNCTIONS_LAMBDA_ENDPOINT` | `default` | **Deprecated.** This is only supported for the `legacy` provider. URL to use as the Lambda service endpoint in Step Functions. By default this is the LocalStack Lambda endpoint. Use default to select the original AWS Lambda endpoint. |
-| `S3_DIR` | | **Deprecated.** This is only supported for the `legacy_v2` provider. Configure a global parent directory that contains all buckets as sub-directories (`S3_DIR=/path/to/root`) or an individual directory that will get mounted as special bucket names (`S3_DIR=/path/to/root/bucket1:bucket1`). Only available for Localstack Pro.
+| Variable | Deprecated since | Example Values | Description |
+| - | - | - | - |
+| `BIGDATA_MONO_CONTAINER` | 0.0.0 | `0`\|`1` (default) | Whether to spin Big Data services inside the LocalStack main container. Glue jobs breaks when using `BIGDATA_MONO_CONTAINER=0`. | 
+| `SKIP_INFRA_DOWNLOADS` | 0.0.0 | | Whether to skip downloading additional infrastructure components (e.g., specific Elasticsearch versions) |
+| `STEPFUNCTIONS_LAMBDA_ENDPOINT` | 3.0.0 | `default` | This is only supported for the `legacy` provider. URL to use as the Lambda service endpoint in Step Functions. By default this is the LocalStack Lambda endpoint. Use default to select the original AWS Lambda endpoint. |
+| `S3_DIR` | 3.0.0 | | This is only supported for the `legacy_v2` provider. Configure a global parent directory that contains all buckets as sub-directories (`S3_DIR=/path/to/root`) or an individual directory that will get mounted as special bucket names (`S3_DIR=/path/to/root/bucket1:bucket1`). Only available for Localstack Pro.
```

Comment:
Inconsistent ordering. Consider ordering with the most-recent deprecation/removal first and then alphabetically. This would highlight the most relevant changes at the top.

Reply:
Yeah that's the plan, just still a kinda raw draft :+1: 

Not sure yet what's going to ultimately be the SSOT. Both options right now don't seem super appealing, especially since we're splitting configs between community and -ext as well.

Reply:
We have two use cases,
• discover options for a specific service (current focus overall)
• discover migration path for deprecations by release (current draft for deprecations/legacy)

I kinda liked having deprecations inline (ie at original place) with a prominent migration path in bold and then collect legacy variables at the bottom. Deprecations are easy to spot when working with service X and easy to find because it does not suddenly disappear to the bottom.


Ultimately, we might want to consider a data-driven approach giving more flexibility in terms of filtering and grouping...

Reply:
I'm fine with integrating the deprecations back into the main list(s) and only keeping legacy in its own "dump" below. The deprecations should theoretically not grow too big in size anyway since every major release should move a big bunch of them to the legacy section.